src/v0/destinations/intercom/util.js

```diff
@@ -0,0 +1,32 @@
+const { MetadataTypes } = require('./config');
+
+/**
+ * Separates reserved metadata from rest of the metadata based on the metadata types
+ * ref:- https://developers.intercom.com/intercom-api-reference/v1.4/reference/event-metadata-types
+ * @param {*} metadata
+ * @returns
+ */
+function separateReservedAndRestMetadata(metadata) {
```

Comment:
Can we improve the names to `flattenedMetadata` and `unflattenedMatadata` if reserved keyword is not crucial?